### PR TITLE
machined: prefer pinning clients via pidfd when creating machine, rat…

### PR DESCRIPTION
**Pull request by poettering**
…her than PID

### CODE REVIEW

File: src/machine/machined-dbus.c [hunk 1]
```diff
@@ -284,21 +290,19 @@ static int method_create_or_register_machine(Manager *manager, sd_bus_message *m
...
-                r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID, &creds);
+                r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID|SD_BUS_CREDS_PIDFD, &creds);
                 if (r < 0)
-                        return r;
-
-                assert_cc(sizeof(uint32_t) == sizeof(pid_t));
+                        return sd_bus_error_set_errnof(error, r, "Failed to acquire client credentials: %m");
 
-                r = sd_bus_creds_get_pid(creds, (pid_t*) &leader);
+                r = bus_creds_get_pidref(creds, &pidref);
```
Comment (YHNdnzj) [inline src/machine/machined-dbus.c]: bus_query_sender_pidref?

File: src/machine/machined-dbus.c [hunk 2]
```diff
@@ -288,23 +288,15 @@ static int method_create_or_register_machine(
                 return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Root directory must be empty or an absolute path");
 
         if (leader == 0) {
-                _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
-
-                r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID, &creds);
+                r = bus_query_sender_pidref(message, &pidref);
```
Comment (bluca) [inline src/machine/machined-dbus.c:291]: Doesn't this require a fallback in case the option is not available?

Comment (YHNdnzj) [reply src/machine/machined-dbus.c:291]: Hmm? What option specifically?

**Review by YHNdnzj — APPROVED** [review]

File: src/machine/machined-dbus.c (hunk 2 above)
Comment (bluca) [reply src/machine/machined-dbus.c:291]: for the pidfd handling

Comment (poettering) [reply src/machine/machined-dbus.c:291]: there's already code in place to handle pidfd being unavailable. see bus_creds_get_pidref()

**Review by bluca — APPROVED** [review]

**Pull request merged**